src/app/collective-rewards/builder/LastCycleRewards.tsx

```diff
@@ -0,0 +1,78 @@
+import { formatBalanceToHuman } from '@/app/user/Balances/balanceUtils'
+import { Address } from 'viem'
+import { usePricesContext } from '@/shared/context/PricesContext'
+import { FC, useEffect } from 'react'
+import { useAlertContext } from '@/app/providers'
+import {
+  NotifyRewardEventLog,
+  useGetNotifyRewardLogs,
+} from '@/app/collective-rewards/hooks/useGetNotifyRewardLogs'
+import { Cycle, useCycleContext } from '@/app/collective-rewards/CycleContext'
+import { MetricsCardMultipleRowsWithSpinner } from '@/app/collective-rewards/components/MetricsCardMultipleRows'
+import { getLastCycleRewards } from '@/app/collective-rewards/builder/utils/getLastCycleRewards'
+import { formatMetrics } from '@/app/collective-rewards/builder/utils/formatMetrics'
+
+type LastCycleRewardsProps = {
+  gauge: Address
+  currency?: string
+  data: {
+    [token: string]: {
+      symbol: string
+      address: Address
+    }
+  }
+}
+
+const useHandleErrors = (errors: { error: Error | null; title: string }[]) => {
```

Comment:
@RootstockCollective/tokenomics I think that we can re-use this, but I'm not sure where to put it :)